org.eclipse.dawnsci.analysis.dataset.test/src/org/eclipse/dawnsci/analysis/dataset/StreamTest.java

```diff
@@ -0,0 +1,159 @@
+		doTest(lzStream,original);
+	}
+
+	private void doTest(Stream<IDataset> stream, Dataset original) {
+		List<Double> collect = stream.map(d -> d.max().doubleValue()).collect(Collectors.toList());
+		Dataset d = DatasetFactory.createFromList(collect);
+		Assert.assertTrue(Comparisons.equalTo(original, d).all());
+	}
+
+	private static IDataset safeSlice(ILazyDataset lz) {
+		try {
+			return lz.getSlice();
+		} catch (DatasetException e) {
+			throw new RuntimeException("Couldn't slice data");
+		}
+	}
+
+	@Test
+	public void iterateImagesNDStream() throws DatasetException {
+		
+		final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+		
+		// That is more like it!
+		SliceStreamSupport.stream(lz, 2,3).forEach(image -> assertTrue(Arrays.equals(new int[]{1,1,100, 100}, image.getShape())));
+	}
+
+	@Test
+	public void iterateImageSumStream1() throws DatasetException {
+		
+		final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+		
+		List<Number> maxes = SliceStreamSupport.sliceStream(lz, 2, 3).map(set->set.max()).collect(Collectors.toList());
+		assertEquals(64*64, maxes.size());
+	}
+
+	@Test
+	public void iterateImageSumStream2() throws DatasetException {
+		
+		final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+		
+		List<Number> maxes = SliceStreamSupport.sliceStream(lz, 0, 1).map(set->set.max()).collect(Collectors.toList());
+		assertEquals(100*100, maxes.size());
+	}
+
+	@Test
+	public void iterateImageSumStream3() throws DatasetException {
+		
+		final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+		
+		List<Number> maxes = SliceStreamSupport.sliceStream(lz, 0, 2).map(set->set.max()).collect(Collectors.toList());
+		assertEquals(64*100, maxes.size());
+	}
+
+	@Test
+	public void iterateImageSumStream1DotParallel() throws DatasetException {
+		
+		final ILazyDataset lz = DatasetFactory.ones(64, 64, 100, 100);		
+		
+		List<Number> maxes = SliceStreamSupport.sliceStream(lz, 2, 3).parallel().map(set->set.max()).collect(Collectors.toList());
+		assertEquals(64*64, maxes.size());
+	}
+
+	@Test
+	public void iterateImageSumStream2DotParallel() throws DatasetException {
+		
+		final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+		
+		List<Number> maxes = SliceStreamSupport.sliceStream(lz, 0, 1).parallel().map(set->set.max()).collect(Collectors.toList());
+		assertEquals(100*100, maxes.size());
+	}
+
+
+	@Test
+	public void iterateImageSumStream1BooleanParallel() throws DatasetException {
+		
+		final ILazyDataset lz = DatasetFactory.ones(64, 64, 100, 100);		
+		
+		List<Number> maxes = SliceStreamSupport.sliceStream(lz, true, 2, 3).map(set->set.max()).collect(Collectors.toList());
+		assertEquals(64*64, maxes.size());
+	}
+
+	@Test
+	public void iterateImageSumStream2BooleanParallel() throws DatasetException {
+		
+		final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+		
+		List<Number> maxes = SliceStreamSupport.sliceStream(lz, true, 0, 1).map(set->set.max()).collect(Collectors.toList());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1612)